### PR TITLE
Allow spaces and special chars in phone number username

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

https://github.com/Expensify/App/issues/6227#issuecomment-962373136

### Fixed Issues

$ https://github.com/Expensify/App/issues/6227

### Tests | QA Steps
1. Navigate to NewDot
2. Input a phone number as login using any of symbols `-`, `(`, `)` or spaces (e.g. +1 (222) 222 2222)

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Code looks good to me. But let's not forget to place this same logic in the request a call modal,
image

**Steps**: Go to Concierge chat -> Click the phone icon on the top right -> Enter a valid phone number with spaces and special characters. Confirm it works fine.

**Comment by parasharrajat** [comment]
@chiragsalian Added the same for RequestCallPage as well.

File: src/pages/RequestCallPage.js [hunk 1]
```diff
@@ -106,13 +106,13 @@ class RequestCallPage extends Component {
             Growl.error(this.props.translate('requestCallPage.growlMessageNoPersonalPolicy'), 3000);
             return;
         }
-
+        const phoneNumber = this.state.phoneNumber.replace(/((?!\n)[()-\s\t])/g, '');
```
Comment (chiragsalian) [inline src/pages/RequestCallPage.js:109]: The regex in this file has been repeated 4 times. It seems pretty useful as well so how about moving it to a lib file and calling it via a method like `getPhoneNumberWithoutSpecialChars` that would do the `replace` as well, or moving the regex to CONST. This way we can DRY up the code.

**Review by chiragsalian — CHANGES_REQUESTED** [review]
Code LGTM and works well. Left a minor suggestion for improvement.

File: src/pages/RequestCallPage.js (hunk 1 above)
Comment (parasharrajat) [reply src/pages/RequestCallPage.js]: Done

**Review by chiragsalian — APPROVED** [review]
Nice, thank you for the changes.

**Pull request merged**